cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -204,85 +203,120 @@ def from_rules(
         )
 
     @staticmethod
-    def containers_from_rules(rule: Rules, space: str | None = None) -> dict[str, ContainerApply]:
+    def containers_from_rules(rules: Rules) -> dict[str, ContainerApply]:
         """Create a dictionary of ContainerApply instances from a Rules instance.
 
         Args:
             rule: instance of Rules.`
-            space: Name of the space to place the containers.
 
         Returns:
             Dictionary of ContainerApply instances.
         """
 
-        if space:
-            rule.update_prefix(space)
-
-        class_properties = to_class_property_pairs(rule)
-        return {
-            class_id: ContainerApply(
-                space=rule.metadata.space,
-                external_id=class_id,
-                name=rule.classes[class_id].class_name,
-                description=rule.classes[class_id].description,
-                properties=DataModel.container_properties_from_dict(properties),
-            )
-            for class_id, properties in class_properties.items()
-        }
+        containers: dict[str, ContainerApply] = {}
+        errors: list = []
+
+        for row, property_ in rules.properties.items():
+            if property_.container:
+                container_property = DataModel.container_property_from_rules_property(property_)
+                id_ = property_.container.id
+
+                if id_ not in containers and container_property:
+                    containers[id_] = ContainerApply(
+                        space=property_.container.space,
+                        external_id=property_.container.external_id,
+                        description=rules.classes[property_.container.external_id].description
+                        if property_.container.external_id in rules.classes
+                        and property_.container.space == rules.metadata.space
+                        else None,
+                        name=rules.classes[property_.container.external_id].class_name
+                        if property_.container.external_id in rules.classes
+                        and property_.container.space == rules.metadata.space
+                        else None,
+                        properties={
+                            cast(str, property_.container_property): cast(ContainerProperty, container_property)
+                        },
+                    )
```

Comment:
I like this, it is implemented, with exception that you cannot introduce `ClassVar` since `mypy` throw error, instead in previous PR I've introduce:

```
model_config: ClassVar[ConfigDict] = ConfigDict(use_enum_values=True)
```

which is the way to have this properly implemented.